exceptions.cabal

```
@@ -46,9 +46,14 @@ library
     mtl                        >= 2.2      && < 2.4,
     stm                        >= 2.2      && < 3,
     transformers               >= 0.5.2.0  && < 0.7
-  if impl(ghc)
-    build-depends:
-      template-haskell         >= 2.11     && < 2.25
+
+  -- template-haskell-lift was added as a boot library in GHC-9.16
+  -- once we no longer wish to backport releases to older major releases,
+  -- this conditional can be dropped
+  if impl(ghc < 9.14)
+    build-depends: template-haskell
+  else
+    build-depends: template-haskell-lift >= 0.1 && <0.2
```

A minor thing, but I'd prefer to list the position condition first, i.e.,

```suggestion
  if impl(ghc >= 9.14)
    build-depends: template-haskell-lift >= 0.1 && <0.2
  else
    build-depends: template-haskell
```